src/components/table.ts

```diff
@@ -75,43 +79,42 @@ export module Components {
      * @param {number} col The column in which to add the Component.
      * @returns {Table} The calling Table.
      */
-    public addComponent(component: Component, row: number, col: number): Table {
-
+    public add(component: Component, row: number, col: number) {
       if (component == null) {
         throw Error("Cannot add null to a table cell");
       }
 
-      var currentComponent = this._rows[row] && this._rows[row][col];
+      if (this._components().indexOf(component) === -1) {
+        component._parent(this);
 
-      if (currentComponent) {
-        component = component.above(currentComponent);
-      }
+        var currentComponent = this._rows[row] && this._rows[row][col];
+        if (currentComponent) {
+          component = component.above(currentComponent);
+        }
 
-      if (super.add(component)) {
         this._nRows = Math.max(row + 1, this._nRows);
         this._nCols = Math.max(col + 1, this._nCols);
         this._padTableToSize(this._nRows, this._nCols);
-
         this._rows[row][col] = component;
+
+        if (this._isAnchored) {
+          component.anchor(this._content);
+        }
+        this.redraw();
       }
       return this;
     }
 
-    /**
-     * Removes a Component.
-     * 
-     * @param {Component} component The Component to be removed.
-     */
-    public removeComponent(component: Component) {
-      super.remove(component);
+    protected _remove(component: Component) {
       for (var r = 0; r < this._nRows; r++) {
         for (var c = 0; c < this._nCols; c++) {
           if (this._rows[r][c] === component) {
             this._rows[r][c] = null;
-            return;
+            return true;
```

Comment:
I know we're not hitting on optimization yet, but it might be worth adding a dictionary/hashmap lookup structure { component: (i, j) } to table so we can avoid this n^2 traversal.